src/drawer.js

```diff
@@ -276,63 +285,87 @@ $.Drawer.prototype = /** @lends OpenSeadragon.Drawer.prototype */{
     },
 
     // private
-    saveContext: function() {
+    saveContext: function(useSketch) {
         if (!this.useCanvas) {
             return;
         }
 
-        this.context.save();
+        var context = useSketch ? this.sketchContext : this.context;
+        context.save();
     },
 
     // private
-    restoreContext: function() {
+    restoreContext: function(useSketch) {
         if (!this.useCanvas) {
             return;
         }
 
-        this.context.restore();
+        var context = useSketch ? this.sketchContext : this.context;
+        context.restore();
     },
 
     // private
-    setClip: function(rect) {
+    setClip: function(rect, useSketch) {
         if (!this.useCanvas) {
             return;
         }
 
-        this.context.beginPath();
-        this.context.rect(rect.x, rect.y, rect.width, rect.height);
-        this.context.clip();
+        var context = useSketch ? this.sketchContext : this.context;
+        context.beginPath();
+        context.rect(rect.x, rect.y, rect.width, rect.height);
+        context.clip();
     },
 
     // private
-    drawRectangle: function(rect, fillStyle) {
+    drawRectangle: function(rect, fillStyle, useSketch) {
+        if (!this.useCanvas) {
+            return;
+        }
+
+        var context = useSketch ? this.sketchContext : this.context;
+        context.save();
+        context.fillStyle = fillStyle;
+        context.fillRect(rect.x, rect.y, rect.width, rect.height);
+        context.restore();
+    },
+
+    /**
+     * Blends the sketch canvas in the main canvas.
+     * The sketch canvas is then cleared.
+     * @param {Float} opacity The opacity of the blending.
+     * @returns {undefined}
+     */
+    blendSketch: function(opacity) {
         if (!this.useCanvas) {
             return;
         }
 
         this.context.save();
-        this.context.fillStyle = fillStyle;
-        this.context.fillRect(rect.x, rect.y, rect.width, rect.height);
+        this.context.globalAlpha = opacity;
+        this.context.drawImage(this.sketchCanvas, 0, 0);
         this.context.restore();
+        this.sketchContext.clearRect(0, 0,
+            this.sketchCanvas.width, this.sketchCanvas.height);
     },
 
     // private
-    drawDebugInfo: function( tile, count, i ){
+    drawDebugInfo: function( tile, count, i, useSketch ){
```

Comment:
Seems like maybe it's better for the debug info not to be affected by opacity? I guess it would be a bit more work to draw it after the sketchCanvas has been blended. What do you think?